src/crypto/NoteEncryption.cpp

```diff
@@ -14,24 +14,34 @@ void clamp_curve25519(unsigned char *key)
 
 void KDF(unsigned char *K,
     const uint256 &dhsecret,
-    const uint256 &in_epk,
+    const uint256 &epk,
     const uint256 &pk_enc,
-    unsigned char in_nonce
+    const uint256 &seed,
+    bool nonce_bit
    )
 {
-    unsigned char distinguisher = 0xC2;
+    unsigned char K_tmp[64];
+    CSHA512 hasher;
 
-    CSHA256 hasher;
-    hasher.Write(&distinguisher, 1);
+    // Trim the first byte from seed
+    std::vector<unsigned char> first_248(seed.begin() + 1, seed.end());
+
+    // First bit is the 'nonce' bit or `i`
+    // Second bit is zero
+    first_248[0] &= 0x3f;
+    first_248[0] |= (nonce_bit ? 1 : 0 << 7);
```

Comment:
This is wrong for two reasons:
* ``? :`` has lower precedence than ``<<``, so the ``1`` is not shifted
* but in any case, the bit ordering is little-endian, so it should be the low-order bits that are masked off and replaced.